plugins/net_plugin/net_plugin.cpp

```diff
@@ -1657,20 +1661,23 @@ namespace eosio {
          close();
          return;
       }
-      do_queue_write();
+      do_queue_write(block_num);
    }
 
    // called from connection strand
-   void connection::do_queue_write() {
-      if( !buffer_queue.ready_to_send(connection_id) || closed() )
+   void connection::do_queue_write(block_num_type block_num) {
+      if( !buffer_queue.ready_to_send(connection_id) || closed() ) {
+         if (block_num > 0) {
+            peer_dlog(this, "not sending block yet ${n}", ("n", block_num) );
```

Comment:
It would be more helpful if you can print out the exact reason: not ready_to send or closed.

Why only log when block_num > 0?

Not sure `c->do_queue_write(0);` below is a good choice. Maybe you can use optional if block_num is not always available.

Reply:
Done.